### PR TITLE
Update latest commits

### PR DESCRIPTION
**Pull request by sherlockvn**
Tmail work fine with contacts service!

https://user-images.githubusercontent.com/43041967/233317702-830fe25d-6d04-4c9d-93e7-0d8772250f4c.mp4

https://user-images.githubusercontent.com/43041967/233605159-3a99e8fd-f59e-474f-b77a-af5ed8ce9356.mp4

### CODE REVIEW

**Review by hoangdat — APPROVED** [review]

File: android/src/main/java/flutter/plugins/contactsservice/contactsservice/ContactsServicePlugin.java [hunk 1]
```diff
@@ -537,6 +552,55 @@ private Cursor getCursorForPhone(String phone) {
...
+          }
+        }
+      } catch (java.lang.Exception e) {
+        e.printStackTrace();
+      }
+    }
+
+    if(cursor != null)
+      cursor.close();
+
```
Comment (nqhhdev) [inline android/src/main/java/flutter/plugins/contactsservice/contactsservice/ContactsServicePlugin.java]:
```suggestion
    if (cursor != null)
      cursor.close();

```

File: example/lib/contacts_list_page.dart [hunk 2]
```diff
@@ -22,10 +22,9 @@ class _ContactListPageState extends State<ContactListPage> {
   Future<void> refreshContacts() async {
     // Load without thumbnails initially.
     var contacts = (await ContactsService.getContacts(
-            withThumbnails: false, iOSLocalizedLabels: iOSLocalizedLabels))
-        .toList();
+        withThumbnails: false, iOSLocalizedLabels: iOSLocalizedLabels));
 //      var contacts = (await ContactsService.getContactsForPhone("8554964652"))
-//          .toList();
+//          ;
```
Comment (nqhhdev) [inline example/lib/contacts_list_page.dart:27]: Remove it

File: example/lib/contacts_list_page.dart [hunk 3]
```diff
@@ -51,7 +49,7 @@ class _ContactListPageState extends State<ContactListPage> {
 
   _openContactForm() async {
     try {
-      var contact = await ContactsService.openContactForm(
+      var _ = await ContactsService.openContactForm(
```
Comment (nqhhdev) [inline example/lib/contacts_list_page.dart:52]: Why do you use `_` ?

File: android/src/main/java/flutter/plugins/contactsservice/contactsservice/ContactsServicePlugin.java [hunk 4]
```diff
@@ -280,14 +293,15 @@ public boolean onActivityResult(int requestCode, int resultCode, Intent intent)
...
+          if (intent != null){
         Cursor cursor = contentResolver.query(contactUri, null, null, null, null);
         if (cursor.moveToFirst()) {
           String id = contactUri.getLastPathSegment();
           getContacts("openDeviceContactPicker", id, false, false, false, localizedLabels, this.result);
         } else {
           Log.e(LOG_TAG, "onActivityResult - cursor.moveToFirst() returns false");
           finishWithResult(FORM_OPERATION_CANCELED);
-        }
+        }}else{return true;}
```
Comment (nqhhdev) [inline android/src/main/java/flutter/plugins/contactsservice/contactsservice/ContactsServicePlugin.java]:
```suggestion
        } else {return true}
```

**Review by dab246 — COMMENTED** [review]
Please test more on Android. Thanks.

Here is demo for android

https://user-images.githubusercontent.com/43041967/233891788-8d815712-83cc-4efd-9d82-376a724858cd.mp4

File: android/src/main/java/flutter/plugins/contactsservice/contactsservice/ContactsServicePlugin.java (hunk 1 above)
Comment (sherlockvn) [reply android/src/main/java/flutter/plugins/contactsservice/contactsservice/ContactsServicePlugin.java]: This code is from library, i didn't change anything. So i think we don't have to fix invention error

**Comment by sherlockvn** [comment]
> Please test more on Android. Thanks.



> Please test more on Android. Thanks.

Here is demo for android
https://user-images.githubusercontent.com/43041967/233891673-928f492c-82cd-4cde-a492-9518e22ecd2c.mp4

**Review by nqhhdev — APPROVED** [review]

**Review by dab246 — APPROVED** [review]

**Comment by hoangdat** [comment]
merged. if you don't want to include the change in your product, please notice your pubspec's change

File: android/src/main/java/flutter/plugins/contactsservice/contactsservice/ContactsServicePlugin.java [hunk 5]
```diff
@@ -293,6 +293,7 @@ public boolean onActivityResult(int requestCode, int resultCode, Intent intent)
           return true;
         }
         Uri contactUri = intent.getData();
+          if (intent != null){
```
Comment (hoangdat) [inline android/src/main/java/flutter/plugins/contactsservice/contactsservice/ContactsServicePlugin.java]: this check for nothing?

File: android/src/main/java/flutter/plugins/contactsservice/contactsservice/ContactsServicePlugin.java [hunk 6]
```diff
@@ -302,6 +303,8 @@ public boolean onActivityResult(int requestCode, int resultCode, Intent intent)
           finishWithResult(FORM_OPERATION_CANCELED);
         }
         cursor.close();
+        }else{return true;}
+
```
Comment (hoangdat) [inline android/src/main/java/flutter/plugins/contactsservice/contactsservice/ContactsServicePlugin.java]: reformat this also

File: android/src/main/java/flutter/plugins/contactsservice/contactsservice/ContactsServicePlugin.java (hunk 5 above)
Comment (sherlockvn) [reply android/src/main/java/flutter/plugins/contactsservice/contactsservice/ContactsServicePlugin.java]: yeah, no need for it. removed it.

Comment (hoangdat) [reply android/src/main/java/flutter/plugins/contactsservice/contactsservice/ContactsServicePlugin.java]: no, IMO, this check for 
```
Uri contactUri = intent.getData();
```

**Pull request merged**